### PR TITLE
Feat: Integrate Charge Information Flow

### PR DESCRIPTION
**Pull request by DGajewska**
WATER-2845

* Creates pojoToModel mappers where required
* Adds type to change reason model
* Adds endpoints to get required data for Charge Information flow

### CODE REVIEW

File: src/modules/charge-versions/routes/charge-version-workflows.js [hunk 1]
```diff
@@ -22,7 +22,10 @@ module.exports = {
         scope: [chargeVersionWorkflowEditor, chargeVersionWorkflowApprover]
       },
       validate: {
-        headers
+        headers,
```
Comment (timtamimi) [inline src/modules/charge-versions/routes/charge-version-workflows.js:25]: If it's OK with you, I think we should probably rename headers to something like `headersValidation` or something like that. 

Not important, just a readability thing.

**Review by timtamimi — CHANGES_REQUESTED** [review]
Looks OK, just one error in `src/lib/models/change-reason.js`

File: src/lib/mappers/invoice-account.js [hunk 2]
```diff
@@ -31,7 +34,10 @@ const pojoToModelMapper = createMapper()
     'id',
     'accountNumber'
   )
-  .map('company').to('company', companyMapper.pojoToModel);
+  .map('company').to('company', companyMapper.pojoToModel)
+  .map('dateRange').to('dateRange', dateRangeMapper.pojoToModel)
+  .map('invoiceAccountAddresses').to('invoiceAccountAddresses',
```
Comment (timtamimi) [inline src/lib/mappers/invoice-account.js:39]: I've spotted another thing, but I'm not sure if this is the source of the problem:

The new invoice account workflow fails to render if the invoiceAccountAddresses are null/nonexistent. 

Stack below:
```
Debug: internal, implementation, error 
1|water-api    |     AssertionError [ERR_ASSERTION]: Address expected
1|water-api    |     at assertIsInstanceOf (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/models/validators.js:50:8)
1|water-api    |     at Role.set address [as address] (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/models/role.js:79:5)
1|water-api    |     at Role.fromHash (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/models/model.js:26:17)
1|water-api    |     at createModel (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/mappers/lib/helpers.js:14:16)
1|water-api    |     at crmToModel (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/mappers/role.js:28:31)
1|water-api    |     at Array.map (<anonymous>)
1|water-api    |     at Object.crmToModel (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/mappers/document.js:18:38)
1|water-api    |     at getDocument (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/services/documents-service.js:31:25)
1|water-api    |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
1|water-api    |     at async module.exports.internals.Manager.execute (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/toolkit.js:45:28)
1|water-api    |     at async Object.internals.handler (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/handler.js:46:20)
1|water-api    |     at async exports.execute (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/handler.js:31:20)
1|water-api    |     at async Request._lifecycle (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/request.js:312:32)
1|water-api    |     at async Request._execute (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/request.js:221:9)
1|water-api    | 2020-10-15T11:40:13.193Z - error: 201015/114013.150, [error] message: Address expected, stack: AssertionError [ERR_ASSERTION]: Address expected
1|water-api    |     at assertIsInstanceOf (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/models/validators.js:50:8)
1|water-api    |     at Role.set address [as address] (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/models/role.js:79:5)
1|water-api    |     at Role.fromHash (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/models/model.js:26:17)
1|water-api    |     at createModel (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/mappers/lib/helpers.js:14:16)
1|water-api    |     at crmToModel (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/mappers/role.js:28:31)
1|water-api    |     at Array.map (<anonymous>)
1|water-api    |     at Object.crmToModel (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/mappers/document.js:18:38)
1|water-api    |     at getDocument (/Users/timtamimi/repos/water/water-abstraction-service/src/lib/services/documents-service.js:31:25)
1|water-api    |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
1|water-api    |     at async module.exports.internals.Manager.execute (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/toolkit.js:45:28)
1|water-api    |     at async Object.internals.handler (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/handler.js:46:20)
1|water-api    |     at async exports.execute (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/handler.js:31:20)
1|water-api    |     at async Request._lifecycle (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/request.js:312:32)
1|water-api    |     at async Request._execute (/Users/timtamimi/repos/water/water-abstraction-service/node_modules/@hapi/hapi/lib/request.js:221:9)
1|water-api    | 2020-10-15T11:40:13.194Z - info: 201015/114013.150, [response] http://ip-192-168-68-106.eu-west-2.compute.internal:8001: get /water/1.0/licences/5cbfbfd4-2414-44bd-b819-f5f06cd43c94/valid-documents/2020-10-15 {} 500 (44ms)
```

File: src/modules/charge-versions/routes/charge-version-workflows.js (hunk 1 above)
Comment (DGajewska) [reply src/modules/charge-versions/routes/charge-version-workflows.js:25]: We've called it `headers` across many routes files. So I'm inclined to leave this as is

File: src/lib/models/change-reason.js [hunk 3]
```diff
@@ -3,6 +3,11 @@
 const Model = require('./model');
 const validators = require('./validators');
 
+const changeReasonTypes = [
+  'new_chargeable_charge_version',
+  'new_non_chargeable_charge_reason'
```
[the remark this answers is not in the document]
Comment (DGajewska) [reply src/lib/models/change-reason.js]: That's a typo, thanks for catching it :)

File: src/lib/mappers/invoice-account.js (hunk 2 above)
Comment (DGajewska) [reply src/lib/mappers/invoice-account.js:39]: I think it's in a different mapper, but I have updated the related mappers to this issue to use the new library, so errors shouldn't be thrown if a data point is missing

**Pull request merged**